### ISSUE TITLE
Terraform init ignoring provider section with swift backend

### ISSUE THREAD

**Issue by michalmedvecky** [terraform-provider-openstack/terraform-provider-openstack#373]
When trying to initialize a swift backend like this:

```
$ terraform init -backend-config=be.conf
```

```
$ cat be.conf
container="tf-hashi-container"
```

Terraform ignores the `provider openstack` section and asks for credentials on stdin.

### Terraform Version
Terraform v0.11.7
+ provider.openstack v1.6.0

### Terraform Configuration Files
```hcl
provider "openstack" {
  version     = "=1.6.0"
  user_name   = "mitchell"
  password = "hashimoto"
  tenant_name = "hashicorp_production"
  auth_url = "http://openstack.hashicorp.com:5000/v2.0/"
  region = "RegionOne"
}

resource "openstack_networking_secgroup_v2" "secgroup_1" {
  name        = "secgroup_1"
  description = "My neutron security group"
}

terraform {
  backend "swift" {}
}
```

### Expected Behavior

I'd expect Terraform to successfully initialize (as my credentials are all ok)

### Actual Behavior
```
# terraform init -backend-config=be.conf

Initializing the backend...
auth_url
  The Identity authentication URL.

  Enter a value: ^C
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform init -backend-config=be.conf`

### Important factoids

When I omit the `backend` section (and write statefile to current dir), everything works as expected.

**Comment by michalmedvecky** [comment]
I tried without `-backend-config`, specifying `container` directly in the `backend` section. 

The result is the same.

**Comment by jtopjian** [comment]
@michalmedvecky hard-coded authentication settings between the OpenStack provider and the Swift backend are not shared. The Swift backend configuration is unable to access the OpenStack provider's information.

The only way to share between the two would be to use environment variables. One thing to keep in mind is that since the code for the Swift backend is located in the core Terraform code, it's not something generally kept up to date by the OpenStack Provider maintainers.

**Comment by michalmedvecky** [comment]
That does not make much sense to a user like me. Can you at least document that somewhere?

In every project, I use environment variables for passing OpenStack credentials. But this time, for some reason, it makes more sense to have them in configs :\

**Comment by jtopjian** [comment]
> Can you at least document that somewhere?

Yes. There's some other work going on right now to update the Swift backend. I'll make sure a doc patch is added to explain this somewhere along the way of this work.

Surprisingly, the only documentation I found to explain this behavior with AWS/S3 is:

> When configuring Terraform, use either environment variables or the standard credentials file ~/.aws/credentials to provide the administrator user's IAM credentials within the administrative account to both the S3 backend and to Terraform's AWS provider.

https://www.terraform.io/docs/backends/types/s3.html

**Comment by jtopjian** [comment]
I'm cleaning up and updating some old issues and wanted to provide an update on this.

There are patches in queue for this, but they're waiting on further Terraform 0.12 work first. I'll still leave this issue opened until it's completely fixed.

**Comment by kayrus** [comment]
swift backend is not a part of this repo
